src/main/java/pl/project13/maven/git/GitReleaseNotesMojo.java

```diff
@@ -0,0 +1,110 @@
+package pl.project13.maven.git;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.Files;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.jetbrains.annotations.NotNull;
+import pl.project13.maven.git.log.LoggerBridge;
+import pl.project13.maven.git.log.MavenLoggerBridge;
+import pl.project13.maven.git.release.ReleaseNotes;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+
+/**
+ * This goal can be used to produce release notes between two tags.
+ *
+ * The release notes are comprised of all commit descriptions from the commit at which the first
+ * tag (release tag, possibily) exists and walks the commit graph,
+ * till the second (release) tag is reached.
+ *
+ * Created by pankaj on 10/21/16.
+ */
+@Mojo(name = "releaseNotes", defaultPhase = LifecyclePhase.INITIALIZE, threadSafe = true)
+public class GitReleaseNotesMojo extends GitMojo {
```

Comment:
I'm also thinking if this would not be better served by being it's own plugin - WDYT?
Would you want to release it independently instead? Having a more focused impl could be beneficial (less cruft)